packages/server/lib/browsers/index.coffee

```diff
@@ -48,9 +50,14 @@ getBrowserLauncherByFamily = (family) ->
 isValidPathToBrowser = (str) ->
   path.basename(str) isnt str
 
-ensureAndGetByNameOrPath = (nameOrPath, returnAll = false) ->
-  utils.getBrowsers(nameOrPath)
+ensureAndGetByNameOrPath = (nameOrPath, returnAll = false, browsers = null) ->
+  findBrowsers = if Array.isArray(browsers) then Promise.resolve(browsers) else utils.getBrowsers()
+
+  findBrowsers
   .then (browsers = []) ->
+    debug("searching for browser '%s' among %s",
+      nameOrPath, pluralize("browser", browsers.length, true))
```

Comment:
this seems unnecessary, debug logs should be consistent and not change wording based on what's being logged